first changes (e due)

cigno5 opened the pull request.

mmardegan commented:
Perfetto: essendo tu collaborator, oltre al pull request avviene anche il merge.

mpaglia0 commented:
Questo significa che hai trovato la soluzione che stavi cercando?

Il giorno dom 24 nov 2019 alle ore 16:53 Matteo Mardegan <
notifications@github.com> ha scritto:

> Perfetto: essendo tu collaborator, oltre al pull request avviene anche il
> merge.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/mmardegan/darktable/pull/2?email_source=notifications&email_token=AFCCBVEQYCBR7AJE2SQPDRTQVKPO7A5CNFSM4JQXERX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFAOMNY#issuecomment-557901367>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFCCBVATI7WGMHG5P5URXXDQVKPO7ANCNFSM4JQXERXQ>
> .
>

mpaglia0 commented:
Posso procedere con la traduzione?

Il giorno lun 25 nov 2019 alle ore 11:01 Maurizio Paglia <mpaglia0@gmail.com>
ha scritto:

> Questo significa che hai trovato la soluzione che stavi cercando?
>
> Il giorno dom 24 nov 2019 alle ore 16:53 Matteo Mardegan <
> notifications@github.com> ha scritto:
>
>> Perfetto: essendo tu collaborator, oltre al pull request avviene anche il
>> merge.
>>
>> —
>> You are receiving this because you modified the open/close state.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/mmardegan/darktable/pull/2?email_source=notifications&email_token=AFCCBVEQYCBR7AJE2SQPDRTQVKPO7A5CNFSM4JQXERX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFAOMNY#issuecomment-557901367>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AFCCBVATI7WGMHG5P5URXXDQVKPO7ANCNFSM4JQXERXQ>
>> .
>>
>

mmardegan commented:
In data lunedì 25 novembre 2019 11:07:12 CET, Maurizio Paglia ha scritto:
> Posso procedere con la traduzione?

Certo, certo: scusa se non ti rispondo velocemente. Importante è non perdere 
lavoro e tempo, per ora procede tutto perfettamente, abbiamo tutto allineato e 
si starà presto a fare il pull al repo principale. 
Non sono andato avanti per problemi di tempo. Spero di risolvere presto. 

> 
> Il giorno lun 25 nov 2019 alle ore 11:01 Maurizio Paglia
> <mpaglia0@gmail.com>
 ha scritto:
> 
> 
> > Questo significa che hai trovato la soluzione che stavi cercando?
> >
> >
> >
> > Il giorno dom 24 nov 2019 alle ore 16:53 Matteo Mardegan <
> > notifications@github.com> ha scritto:
> >
> >
> >
> >> Perfetto: essendo tu collaborator, oltre al pull request avviene anche
> >> il
> >> merge.
> >>
> >>
> >>
> >> —
> >> You are receiving this because you modified the open/close state.
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/mmardegan/darktable/pull/2?email_source=notifications
> >> &email_token=AFCCBVEQYCBR7AJE2SQPDRTQVKPO7A5CNFSM4JQXERX2YY3PNVWWK3TUL52H
> >> S4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFAOMNY#issuecomment-55790136
> >> 7>,
 or unsubscribe
> >> <https://github.com/notifications/unsubscribe-auth/AFCCBVATI7WGMHG5P5URXX
> >> DQVKPO7ANCNFSM4JQXERXQ>
 .
> >>
> >>
> >
> >
> 
> 
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/mmardegan/darktable/pull/2#issuecomment-558082883


-- 
Matteo Mardegan
GPG finger print 552BA209B53DE0AB4CE4670FADB2DD4BF784D834
key id F784D834 su https://pgp.key-server.io/

mpaglia0 commented:
Benissimo!
Ero un po' preoccupato e mi sentivo in colpa per il casino che ho fatto con
git... :-(

Il giorno lun 25 nov 2019 alle ore 14:53 Matteo Mardegan <
notifications@github.com> ha scritto:

> In data lunedì 25 novembre 2019 11:07:12 CET, Maurizio Paglia ha scritto:
> > Posso procedere con la traduzione?
>
> Certo, certo: scusa se non ti rispondo velocemente. Importante è non
> perdere
> lavoro e tempo, per ora procede tutto perfettamente, abbiamo tutto
> allineato e
> si starà presto a fare il pull al repo principale.
> Non sono andato avanti per problemi di tempo. Spero di risolvere presto.
>
> >
> > Il giorno lun 25 nov 2019 alle ore 11:01 Maurizio Paglia
> > <mpaglia0@gmail.com>
> ha scritto:
> >
> >
> > > Questo significa che hai trovato la soluzione che stavi cercando?
> > >
> > >
> > >
> > > Il giorno dom 24 nov 2019 alle ore 16:53 Matteo Mardegan <
> > > notifications@github.com> ha scritto:
> > >
> > >
> > >
> > >> Perfetto: essendo tu collaborator, oltre al pull request avviene anche
> > >> il
> > >> merge.
> > >>
> > >>
> > >>
> > >> —
> > >> You are receiving this because you modified the open/close state.
> > >> Reply to this email directly, view it on GitHub
> > >> <
> https://github.com/mmardegan/darktable/pull/2?email_source=notifications
> > >>
> &email_token=AFCCBVEQYCBR7AJE2SQPDRTQVKPO7A5CNFSM4JQXERX2YY3PNVWWK3TUL52H
> > >>
> S4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFAOMNY#issuecomment-55790136
> > >> 7>,
> or unsubscribe
> > >> <
> https://github.com/notifications/unsubscribe-auth/AFCCBVATI7WGMHG5P5URXX
> > >> DQVKPO7ANCNFSM4JQXERXQ>
> .
> > >>
> > >>
> > >
> > >
> >
> >
> >
> > --
> > You are receiving this because you commented.
> > Reply to this email directly or view it on GitHub:
> > https://github.com/mmardegan/darktable/pull/2#issuecomment-558082883
>
>
> --
> Matteo Mardegan
> GPG finger print 552BA209B53DE0AB4CE4670FADB2DD4BF784D834
> key id F784D834 su https://pgp.key-server.io/
>
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/mmardegan/darktable/pull/2?email_source=notifications&email_token=AFCCBVGW73OBTWKED65UJVLQVPKE3A5CNFSM4JQXERX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFCO2JQ#issuecomment-558165286>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFCCBVEUMLONHIPW6ZAWF3DQVPKE3ANCNFSM4JQXERXQ>
> .
>

aurelienpierre committed:
highlights.c: fix a bug in WB undoing

aurelienpierre committed:
highlights.c: fix a bug in OpenCL

loop over 9 elements, 0 is a typo

aurelienpierre committed:
highlights.c: rewrite OpenCL interpolate_and_mask()

Make it look closer to the C version to help debugging

aurelienpierre committed:
highlights.c: manually unroll pixels fetch

helps debugging C vs. OpenCL
speed-up for compilation without -funroll-loops

aurelienpierre committed:
highlights.c: use isotropic color diffusion

anisotropic is too brittle and slower

aurelienpierre committed:
bspline.h: prevent digits cancellation

rewrite the scalar product linearly as to match the OpenCL variant, both in logic and result.

Division by 16 after summation may reduce accuracy of the result (digits cancellation).

Runtimes are unaffected.

aurelienpierre committed:
highlights.c OpenCL: avoid access to float4 vec as array

Sources disagree on how &(float4) can be casted as (*float) and accessed later as array.
There is a chance that the data ordering is not as expected.
In doubt, play it safe. Even if the code looks ugly.

aurelienpierre committed:
highlights.c OpenCL: remove call to native_sqrt

Play safe to ensure consistency with C path.
native_sqrt is implementation-dependent.

aurelienpierre committed:
highlights.c: renormalize ratios before converting to RGB

as suggested by @flannelhead

aurelienpierre committed:
highlights.c: introduce a scale multiplier

reduce the contribution of farther pixels in the reconstruction
as to avoid inpanting dark region into bright clipped highlights

aurelienpierre committed:
highlights.c: fix SIMD loops

dt_aligned_pixels_t are not aligned on 64 bits

aurelienpierre committed:
guided laplacian: fix align issues, soften scale penalty

phweyland committed:
metadata: avoid to apply on hovered image instead of selection

attempt to fixe #11589

aurelienpierre committed:
highlights: guided laplacian: boost color propagation

phweyland committed:
tagging:  avoid to apply on hovered image instead of selection

Nilvus committed:
fix missing padding right and missing icons on some overlays with shortest code

aurelienpierre committed:
Revert "highlights: guided laplacian: boost color propagation"

This reverts commit 62274e951fbe8e7fcadfd652d730e1348d2e0001.

TurboGit committed:
Merge pull request #11863 from phweyland/metadata-hover

metadata: avoid to apply on hovered image instead of selection

aurelienpierre committed:
highlights: fix legacy param bug / introduce first order param

Nilvus committed:
fix and some overlays improvements

jenshannoschwalm committed:
Avoid unnecessary reprocessing the pipeline in highlights

When the module gets out-of-focus we only need to reprocess the pipeline if the
visualizng mode was switched on.

EdgarLux committed:
Spanish translation update

jenshannoschwalm committed:
Avoid unnecessary pipeline reprocessing in demosaic focus_out

Only requires reprocessing if mask button was switched on.

jenshannoschwalm committed:
Avoid pipeline reprocessing while gui_unfocus tone equalizer

Reprocessing is only required if mask was switched on.

TurboGit committed:
Merge pull request #11872 from Nilvus/some-css-fixes

Fix missing padding and icons on some overlays with shortest code

TurboGit committed:
Merge pull request #11817 from aurelienpierre/guided-laplacian-bugfixes

Guided laplacian bugfixes

jenshannoschwalm committed:
Use simpler approach to get mask status as @TurbuGit suggested

jpellegrini committed:
Updates to pt_BR

phweyland committed:
variables: add missing dt_variables_params_destroy() (memory leak)

aurelienpierre committed:
highlights: prevent division by zero

account for rounding errors in clipping threshold
allow larger noise factor

aurelienpierre committed:
color balance : move color space combobox

add a tooltip

jpellegrini committed:
Updates to pt_BR

Commit:
Update Russian translation.

EdgarLux committed:
Spanish translation update

EdgarLux committed:
Spanish translation: revision

quovadit committed:
filtering: custom rating-tooltip (#11848)

stephanh73 committed:
Update Dutch translation.

victoryforce committed:
No space before punctuation marks

TurboGit committed:
Update darktable.pot for translators.

TurboGit committed:
Update French translation.

TurboGit committed:
colorbalancergb.c: Minor string fix.

EdgarLux committed:
Spanish translation: untranslated strings marked as fuzzy for better management

jpellegrini committed:
Updates to pt_BR

mmardegan committed:
start italian translation for dt 4.0

MStraeten committed:
Update de.po

to latest pot file

victoryforce committed:
No space before colons (preset names)

victoryforce committed:
Ukrainian translation update

TurboGit committed:
Update darktable.pot for translators.

TurboGit committed:
Update French translation.

shinoryo1216 committed:
Japanese translation update (#11901)

flannelhead committed:
Add push and pop pragmas in headers where optimization flags are changed

This prevents the attributes from propagating to whatever file includes
the header.

Fixes https://github.com/darktable-org/darktable/issues/11899

flannelhead committed:
diffuse: remove unnecessary fast_guided_filter.h include

mmardegan committed:
Merge branch 'darktable-org:master' into Traduzione_4.0

mmardegan committed:
update italian translation: bovirus requests

mmardegan committed:
update italian translation

dtorop committed:
PDF export: fix crash or blank file

Add a conf entry for bpp with options 8 or 16 bit. Without this, PDF
export defaults to a bpp of 0, which results in writing a 0 byte
image stream. This causes an empty page with deflate compression, or a
crash with ASCIIHex.

Fixes #11846.

EdgarLux committed:
Spanish translation: adding more strings translations

serkan-maker committed:
update tr.po (#11903)

MStraeten committed:
Update de.po

following fast pot file changes

victoryforce committed:
Ukrainian translation update

flannelhead committed:
Consolidate extra optimization options into a single header

Some of the modules and header files apply a set of extra optimization
flags via a #pragma GCC optimize clause. Consolidate them into a single
header file which can be included at the very beginning of a *.c file in
order to apply the same optimizations to all included headers. This will
help the compiler to inline more functions from the headers - if
optimization flags would mismatch, inlining fail and cause perf issues.
See https://github.com/darktable-org/darktable/issues/11899#issuecomment-1138917001
for an explanation.

Some no longer meaningful flags are dropped. This information is based
on GCC 8.5 documentation:

* split-ivs-in-unroller, fp-contract=fast: enabled by default
* fast-math: enabled globally for Release build (except for finite-math-only)
* tree-loop-if-convert: enabled by default if vectorization is enabled
* tree-loop-distribution, loop-interchange, unswitch-loops, predictive-commoning: enabled at O3
* tree-loop-linear, loop-block, loop-strip-mine: same as loop-nest-optimize

In addition, no-strict-aliasing is dropped - hopefully there are no
strict aliasing issues. At least GCC doesn't warn about any when
compiling with -Wstrict-aliasing. no-strict-aliasing would prevent some
optimizations.

GCC optimization pragmas are removed from three other header files that
used to have them and accidentally caused the optimizations to propagate
to *.c files that included them. It's better to include the new header
at the beginning of a *.c file. The necessary includes are added.

shinoryo1216 committed:
Japanese translation update (#11912)

parafin committed:
fix a crash caused by #11572, fixes #11894

parafin committed:
macOS packaging: drop support for Xcode < 12.4 and macOS < 10.14, fixes #11687

parafin committed:
macOS packaging: remove what seems to be unneeded command

parafin committed:
macOS packaging: fix lensfun update script installation

parafin committed:
CI: try to fix latest macOS-12 build

TurboGit committed:
integration: Update sub-module.

TurboGit committed:
extra_optimizations.h: Minor code reformatting.

mmardegan committed:
Merge branch 'darktable-org:master' into Traduzione_4.0

cigno5 committed:
first changes

cigno5 committed:
another batch

cigno5 committed:
another batch 2

cigno5 committed:
batch 3

cigno5 committed:
batch 4

cigno5 committed:
batch 5

mmardegan committed:
update italian translation
